### PR TITLE
:sparkles:Add missing Fields to RosaControlPlane - tags, etcdEncryption

### PR DESCRIPTION
**Pull request by xiangjingli**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Adding the missing RosaControlPlan API fields;
Tags,
EtcdEncryption,
EtcdEncryptionKMSArn

Also removing the accountID and creatorARN from ROSA cluster templates as it is not necessary to declare them explicitly in the rosaControlPlane spec along the recent change. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/4737

**Special notes for your reviewer**:
There is another PR for adding different fields to RosaControlplan delivered by @serngawy 
https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4780

I will rebase the PR after the 4780 PR is merged.

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] includes [emojis](https://github.com/kubernetes-sigs/kubebuilder-release-tools?tab=readme-ov-file#kubebuilder-project-versioning)
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by xiangjingli** [comment]
@muraee @stevekuznetsov @serngawy FYI Please review the new PR I created, where I addressed all your comments

File: controlplane/rosa/controllers/rosacontrolplane_controller.go [hunk 1]
```diff
@@ -575,6 +594,23 @@ func validateControlPlaneSpec(rosaClient *rosa.RosaClient, rosaScope *scope.ROSA
 		return &message, nil
 	}
 
+	// validate Etcd Encryption KMS Arn
+	err = kmsArnRegexpValidator.ValidateKMSKeyARN(&rosaScope.ControlPlane.Spec.EtcdEncryptionKMSArn)
+	if err != nil {
+		return nil, err
```
Comment (muraee) [inline controlplane/rosa/controllers/rosacontrolplane_controller.go:600]: the function should return a failure message if something is invalid. returning an error means there was a real error during the validation process, like a failed API call.

Comment (muraee) [reply controlplane/rosa/controllers/rosacontrolplane_controller.go:600]: this validation can also be moved to the webhook similar to tags.

Comment (xiangjingli) [reply controlplane/rosa/controllers/rosacontrolplane_controller.go]: The validation function has been moved to the webhook as suggested.

File: controlplane/rosa/api/v1beta2/rosacontrolplane_types.go [hunk 2]
```diff
@@ -79,6 +79,15 @@ type RosaControlPlaneSpec struct { //nolint: maligned
 	// IdentityRef is a reference to an identity to be used when reconciling the managed control plane.
 	// If no identity is specified, the default identity for this controller will be used.
 	IdentityRef *infrav1.AWSIdentityReference `json:"identityRef,omitempty"`
+
+	// User-defined tags for AWS resources
+	// +optional
+	Tags map[string]string `json:"tags,omitempty"`
```
[the remark this answers is not in the document]
Comment (xiangjingli) [reply controlplane/rosa/api/v1beta2/rosacontrolplane_types.go]: updated as suggested

[the remark this answers is not in the document]
Comment (muraee) [reply controlplane/rosa/api/v1beta2/rosacontrolplane_types.go]: the field needs to be called `AdditionalTags` for this to work, it is also better for consistency.

File: controlplane/rosa/controllers/rosacontrolplane_controller.go [hunk 3]
```diff
@@ -575,6 +593,17 @@ func validateControlPlaneSpec(rosaClient *rosa.RosaClient, rosaScope *scope.ROSA
...
+	// sanitize Tags
+	tagsList := map[string]string{}
+
+	if len(rosaScope.ControlPlane.Spec.AdditionalTags) > 0 {
+		for tagKey, tagVal := range rosaScope.ControlPlane.Spec.AdditionalTags {
+			tagsList[strings.TrimSpace(tagKey)] = strings.TrimSpace(tagVal)
+		}
+	}
+
+	rosaScope.ControlPlane.Spec.AdditionalTags = tagsList
```
Comment (muraee) [inline controlplane/rosa/controllers/rosacontrolplane_controller.go:605]: this is not needed with the validation in the webhook.

Comment (xiangjingli) [reply controlplane/rosa/controllers/rosacontrolplane_controller.go]: removed

File: controlplane/rosa/api/v1beta2/rosacontrolplane_types.go [hunk 4]
```diff
@@ -79,6 +79,15 @@ type RosaControlPlaneSpec struct { //nolint: maligned
...
 	// If no identity is specified, the default identity for this controller will be used.
 	IdentityRef *infrav1.AWSIdentityReference `json:"identityRef,omitempty"`
+
+	// User-defined tags for AWS resources
+	// +optional
+	AdditionalTags infrav1.Tags `json:"additionalTags,omitempty"`
+
+	// The etcd encryption kms key ARN is the key used to encrypt etcd.
+	// It is only allowed for hosted cp and it needs to be pre-created in AWS KMS with tag red-hat:true.
+	// +optional
```
Comment (stevekuznetsov) [inline controlplane/rosa/api/v1beta2/rosacontrolplane_types.go:89]: If we ditch the webhook, this is simply:

```
// +kubebuilder:validation:XValidation:rule="self.matches('^arn:aws[\\w-]*:kms:[\\w-]+:\\d{12}:key\\/(mrk-[0-9a-f]{32}$|[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$)')", message="etcdEncryptionKMSArn must be a valid encryption key ARN"
```

**Comment by stevekuznetsov** [comment]
Superseded by #4832 
/close

**Pull request closed without merging**